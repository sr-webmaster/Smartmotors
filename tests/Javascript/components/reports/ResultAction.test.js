import { createLocalVue, shallowMount, mount } from '@vue/test-utils'
import Component from '~/components/reports/ResultAction.vue'

describe('ResultAction', () => {
  // Component instance "under test"
  let localVue
  describe('Snapshots', () => {
    test('defaults', () => {
      localVue = createLocalVue()
      const wrapper = mount(Component, {
        localVue: localVue,
        propsData: {
          dateBefore: 'Oct 7',
          dateAfter: 'Oct 13'
        }
      })
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('Methods', () => {
    beforeEach(() => {
      localVue = createLocalVue()
    })

    test('dataVisibility function emitted data-visibility', () => {
      const wrapper = shallowMount(Component)
      wrapper.vm.dataVisibility()
      expect(wrapper.emitted()['data-visibility']).toBeTruthy()
    })

    describe('saveReport()', () => {
      test('emitted save-report', () => {
        const wrapper = shallowMount(Component)
        wrapper.vm.saveReport()
        expect(wrapper.emitted()['save-report']).toBeTruthy()
      })
    })

    test('exportReport function emitted export-report', () => {
      const wrapper = shallowMount(Component)
      wrapper.vm.exportReport('csv')
      expect(wrapper.emitted()['export-report']).toBeTruthy()
      expect(wrapper.emitted()['export-report'][0][0]).toEqual('csv')
    })
  })
})
