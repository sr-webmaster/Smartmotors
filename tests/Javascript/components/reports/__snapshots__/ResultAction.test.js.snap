// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultAction Snapshots defaults 1`] = `
<div
  class="container pa-0"
  dateafter="Oct 13"
  datebefore="Oct 7"
>
  <div
    class="layout mb-2"
  >
    <h3
      class="headline font-weight-bold"
    >
      
      Financial Report Results
    
    </h3>
  </div>
   
  <div
    class="layout flex wrap"
  >
    <div
      class="flex xs6 sm2 px-2"
    >
      <button
        class="v-btn v-btn--block v-btn--disabled v-btn--round theme--light"
        disabled="disabled"
        type="default"
        xs6=""
      >
        <div
          class="v-btn__content"
        >
          
        Advanced filters
      
        </div>
      </button>
    </div>
     
    <div
      class="flex xs6 sm2 px-2"
    >
      <div>
        <button
          class="sm-ReportsDataVisibility__activator v-btn v-btn--round theme--light primary"
          type="button"
        >
          <div
            class="v-btn__content"
          >
            
    Data Visibility
  
          </div>
        </button>
         
        <aside
          class="v-navigation-drawer v-navigation-drawer--absolute v-navigation-drawer--close v-navigation-drawer--right v-navigation-drawer--temporary theme--light"
          style="height: 100%; margin-top: 0px; transform: translateX(300px); width: 300px;"
        >
          <div
            class="container sm-ReportsDataVisibility__container"
          >
            <h2
              class="mb-4"
            >
              
        Data Visibility
      
            </h2>
             
            <h3
              class="sm-ReportsDataVisibility__sectionTitle"
            >
              
        Active
      
            </h3>
             
            <span
              class="list-group"
            />
             
            <h3
              class="sm-ReportsDataVisibility__sectionTitle"
            >
              
        Inctive
      
            </h3>
             
            <div
              class="v-input v-text-field theme--light"
            >
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      aria-hidden="true"
                      class="v-label theme--light"
                      style="left: 0px; position: absolute;"
                    >
                      Search Parameters
                    </label>
                    <input
                      aria-label="Search Parameters"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <div
                      class="v-messages__wrapper"
                    />
                  </div>
                </div>
              </div>
            </div>
              
            <div
              class="layout"
            >
              <button
                class="v-btn v-btn--block v-btn--flat theme--light secondary--text"
                type="button"
              >
                <div
                  class="v-btn__content"
                >
                  
          Cancel
        
                </div>
              </button>
               
              <button
                class="v-btn v-btn--block theme--light primary"
                type="button"
              >
                <div
                  class="v-btn__content"
                >
                  
          Apply
        
                </div>
              </button>
            </div>
          </div>
          <div
            class="v-navigation-drawer__border"
          />
        </aside>
      </div>
    </div>
     
    <div
      class="flex xs6 sm2 px-2"
    >
      <div
        class="v-dialog__container"
        style="display: inline-block;"
      >
        <button
          class="sm-ReportsSaveModal__activator v-btn v-btn--round theme--light primary"
          type="button"
        >
          <div
            class="v-btn__content"
          >
            
      Save report
    
          </div>
        </button>
        <div
          class="v-dialog__content"
          style="z-index: 0;"
          tabindex="-1"
        >
          <div
            class="v-dialog"
            style="max-width: 400px; display: none;"
          >
             
            <div
              class="v-card v-sheet theme--light"
            >
              <nav
                class="v-toolbar theme--light"
                style="margin-top: 0px; padding-right: 0px; padding-left: 0px; transform: translateY(0px);"
              >
                <div
                  class="v-toolbar__content"
                  style="height: 64px;"
                >
                  <div
                    class="v-toolbar__title"
                  >
                    Save Report
                  </div>
                   
                  <div
                    class="spacer"
                  />
                   
                  <div
                    class="v-toolbar__items"
                  >
                    <button
                      class="v-btn v-btn--icon theme--light"
                      type="button"
                    >
                      <div
                        class="v-btn__content"
                      >
                        <i
                          aria-hidden="true"
                          class="v-icon material-icons theme--light"
                        >
                          close
                        </i>
                      </div>
                    </button>
                  </div>
                </div>
              </nav>
               
              <div
                class="v-card__text"
              >
                <form
                  class="v-form"
                  novalidate="novalidate"
                >
                  <div
                    class="v-input v-text-field theme--light"
                  >
                    <div
                      class="v-input__control"
                    >
                      <div
                        class="v-input__slot"
                      >
                        <div
                          class="v-text-field__slot"
                        >
                          <label
                            aria-hidden="true"
                            class="v-label theme--light"
                            style="left: 0px; position: absolute;"
                          >
                            Report name
                          </label>
                          <input
                            aria-label="Report name"
                            type="text"
                          />
                        </div>
                      </div>
                      <div
                        class="v-text-field__details"
                      >
                        <div
                          class="v-messages theme--light"
                        >
                          <div
                            class="v-messages__wrapper"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                   
                  <div
                    class="v-input v-text-field v-select theme--light"
                    role="combobox"
                  >
                    <div
                      class="v-input__control"
                    >
                      <div
                        class="v-input__slot"
                      >
                        <div
                          class="v-select__slot"
                        >
                          <label
                            aria-hidden="true"
                            class="v-label theme--light"
                            style="left: 0px; position: absolute;"
                          >
                            Modifier 1
                          </label>
                          <div
                            class="v-select__selections"
                          >
                            <input
                              aria-label="Modifier 1"
                              aria-readonly="false"
                              autocomplete="on"
                              readonly="readonly"
                              type="text"
                            />
                          </div>
                          <div
                            class="v-input__append-inner"
                          >
                            <div
                              class="v-input__icon v-input__icon--append"
                            >
                              <i
                                aria-hidden="true"
                                class="v-icon material-icons theme--light"
                              >
                                arrow_drop_down
                              </i>
                            </div>
                          </div>
                        </div>
                        <div
                          class="v-menu"
                        >
                          <div
                            class="v-menu__content theme--light "
                            style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                          >
                            <div
                              class="v-select-list v-card theme--light"
                            >
                              <div
                                class="v-list theme--light"
                                role="list"
                              >
                                <div
                                  role="listitem"
                                >
                                  <div
                                    class="v-list__tile theme--light"
                                  >
                                    <div
                                      class="v-list__tile__content"
                                    >
                                      <div
                                        class="v-list__tile__title"
                                      >
                                        No data available
                                      </div>
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class="v-text-field__details"
                      >
                        <div
                          class="v-messages theme--light"
                        >
                          <div
                            class="v-messages__wrapper"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                   
                  <div
                    class="v-input v-text-field v-select theme--light"
                    role="combobox"
                  >
                    <div
                      class="v-input__control"
                    >
                      <div
                        class="v-input__slot"
                      >
                        <div
                          class="v-select__slot"
                        >
                          <label
                            aria-hidden="true"
                            class="v-label theme--light"
                            style="left: 0px; position: absolute;"
                          >
                            Modifier 2
                          </label>
                          <div
                            class="v-select__selections"
                          >
                            <input
                              aria-label="Modifier 2"
                              aria-readonly="false"
                              autocomplete="on"
                              readonly="readonly"
                              type="text"
                            />
                          </div>
                          <div
                            class="v-input__append-inner"
                          >
                            <div
                              class="v-input__icon v-input__icon--append"
                            >
                              <i
                                aria-hidden="true"
                                class="v-icon material-icons theme--light"
                              >
                                arrow_drop_down
                              </i>
                            </div>
                          </div>
                        </div>
                        <div
                          class="v-menu"
                        >
                          <div
                            class="v-menu__content theme--light "
                            style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                          >
                            <div
                              class="v-select-list v-card theme--light"
                            >
                              <div
                                class="v-list theme--light"
                                role="list"
                              >
                                <div
                                  role="listitem"
                                >
                                  <div
                                    class="v-list__tile theme--light"
                                  >
                                    <div
                                      class="v-list__tile__content"
                                    >
                                      <div
                                        class="v-list__tile__title"
                                      >
                                        No data available
                                      </div>
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class="v-text-field__details"
                      >
                        <div
                          class="v-messages theme--light"
                        >
                          <div
                            class="v-messages__wrapper"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </form>
              </div>
               
              <div
                class="v-card__actions"
              >
                <button
                  class="v-btn v-btn--block v-btn--flat theme--light secondary--text"
                  type="button"
                >
                  <div
                    class="v-btn__content"
                  >
                    
        Cancel
      
                  </div>
                </button>
                 
                <button
                  class="v-btn v-btn--block theme--light primary"
                  type="button"
                >
                  <div
                    class="v-btn__content"
                  >
                    
        Save
      
                  </div>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="flex xs6 sm2 px-2"
    >
      <div
        class="v-dialog__container"
        style="display: inline-block;"
      >
        <button
          class="sm-ReportsExportModal__activator v-btn v-btn--round theme--light primary"
          type="button"
        >
          <div
            class="v-btn__content"
          >
            
      Export
    
          </div>
        </button>
        <div
          class="v-dialog__content"
          style="z-index: 0;"
          tabindex="-1"
        >
          <div
            class="v-dialog"
            style="max-width: 400px; display: none;"
          >
             
            <div
              class="v-card v-sheet theme--light"
            >
              <nav
                class="v-toolbar theme--light"
                style="margin-top: 0px; padding-right: 0px; padding-left: 0px; transform: translateY(0px);"
              >
                <div
                  class="v-toolbar__content"
                  style="height: 64px;"
                >
                  <div
                    class="v-toolbar__title"
                  >
                    Export Report
                  </div>
                   
                  <div
                    class="spacer"
                  />
                   
                  <div
                    class="v-toolbar__items"
                  >
                    <button
                      class="v-btn v-btn--icon theme--light"
                      type="button"
                    >
                      <div
                        class="v-btn__content"
                      >
                        <i
                          aria-hidden="true"
                          class="v-icon material-icons theme--light"
                        >
                          close
                        </i>
                      </div>
                    </button>
                  </div>
                </div>
              </nav>
               
              <div
                class="v-card__text"
              >
                <form
                  class="v-form"
                  novalidate="novalidate"
                >
                  <div
                    class="v-input v-input--selection-controls v-input--radio-group v-input--radio-group--column v-input--is-label-active v-input--is-dirty theme--light"
                  >
                    <div
                      class="v-input__control"
                    >
                      <div
                        class="v-input__slot"
                        style="height: auto;"
                      >
                        <div
                          class="v-input--radio-group__input"
                          role="radiogroup"
                        >
                          <div
                            class="v-radio theme--light"
                          >
                            <div
                              class="v-input--selection-controls__input"
                            >
                              <input
                                aria-checked="false"
                                aria-label="PDF"
                                name="v-radio-58"
                                role="radio"
                                type="radio"
                                value="pdf"
                              />
                              <div
                                class="v-input--selection-controls__ripple"
                              />
                              <i
                                aria-hidden="true"
                                class="v-icon material-icons theme--light"
                              >
                                radio_button_unchecked
                              </i>
                            </div>
                            <label
                              aria-hidden="true"
                              class="v-label theme--light"
                              style="left: 0px; position: relative;"
                            >
                              PDF
                            </label>
                          </div>
                           
                          <div
                            class="v-radio theme--light"
                          >
                            <div
                              class="v-input--selection-controls__input"
                            >
                              <input
                                aria-checked="false"
                                aria-label="CSV"
                                name="v-radio-58"
                                role="radio"
                                type="radio"
                                value="csv"
                              />
                              <div
                                class="v-input--selection-controls__ripple"
                              />
                              <i
                                aria-hidden="true"
                                class="v-icon material-icons theme--light"
                              >
                                radio_button_unchecked
                              </i>
                            </div>
                            <label
                              aria-hidden="true"
                              class="v-label theme--light"
                              style="left: 0px; position: relative;"
                            >
                              CSV
                            </label>
                          </div>
                        </div>
                      </div>
                      <div
                        class="v-messages theme--light"
                      >
                        <div
                          class="v-messages__wrapper"
                        />
                      </div>
                    </div>
                  </div>
                </form>
              </div>
               
              <div
                class="v-card__actions"
              >
                <button
                  class="v-btn v-btn--block v-btn--flat theme--light secondary--text"
                  type="button"
                >
                  <div
                    class="v-btn__content"
                  >
                    
        Cancel
      
                  </div>
                </button>
                 
                <button
                  class="v-btn v-btn--block theme--light primary"
                  type="button"
                >
                  <div
                    class="v-btn__content"
                  >
                    
        Export
      
                  </div>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
